File: core/src/ru/myitschool/dcrawler/entities/skills/patterns/EnemyTargetPattern.java
package ru.myitschool.dcrawler.entities.skills.patterns;

import ru.myitschool.dcrawler.dungeon.DungeonCell;
import ru.myitschool.dcrawler.dungeon.DungeonMap;
import ru.myitschool.dcrawler.entities.skills.Skill;
import ru.myitschool.dcrawler.entities.skills.Target;

public class EnemyTargetPattern extends TargetPattern {
    public EnemyTargetPattern(Skill skill) {
        super(skill);
    }

    @Override
    public Target createTarget(Target target) {
        DungeonCell cell = DungeonMap.getCell(target.getX(), target.getY());
        if (cell.hasEntity() && cell.getEntity().isEnemy()){
            return target;
        }
        return null;
    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/skills/patterns/EnemyTargetPattern.kt
package com.wolg_vlad.dcrawler.entities.skills.patterns

import com.wolg_vlad.dcrawler.dungeon.DungeonMap.Companion.getCell
import com.wolg_vlad.dcrawler.entities.skills.Skill
import com.wolg_vlad.dcrawler.entities.skills.Target

class EnemyTargetPattern(skill: Skill) : TargetPattern(skill) {
    override fun createTarget(target: Target): Target {
        return target
    }

    override fun isValid(target: Target): Boolean {
        val cell = getCell(target.checkX, target.checkY)
        return cell?.entity?.isEnemy?:false
    }
}